src/opentimelineio/clip.cpp

```diff
@@ -19,33 +25,74 @@ Clip::~Clip()
 MediaReference*
 Clip::media_reference() const noexcept
 {
-    return _media_reference;
+    auto active = _media_references.find(_active_media_reference);
+    return active == _media_references.end() || !active->second
+               ? nullptr
+               : active->second;
+}
+
+Clip::MediaReferences
+Clip::media_references() const noexcept
```

Comment:
Although I think it would be more useful to be able to return a mutable map the Python bindings make this very tricky to do.  The only way to do it currently is to use an AnyDictionary (like the `metadata` property on a SerializableObject), but I don't think that's wise to do here as we don't want the value of the map to be anything other than a MediaReference. 

Thus, I just use a simple getter/setter instead so minimal Python glue is needed.  The only real requirement is to convert between Retainer<> and raw pointers, which is the reason these methods don't just do a simple `=`.

Reply:
We could also consider doing the mutable mapping wrapping of these functions in the python layer for that api convenience.
I can take a pass at trying to implement that.

Reply:
A mutable mapping is what I originally wanted to do and took a brief shot at it, but it is indeed a little tricky.  If it were available in the api I would certainly use it! 

We discussed this a bit on slack already in case you didn't see it:
https://academysoftwarefdn.slack.com/archives/CMQ9J4BQC/p1647149228172289